Fixing dispatch/method specialization error

cgeoga opened the pull request:
Hi @dpo and company---

Thanks for the great package! As I've been tinkering with it, I've caught an issue that the special `ldiv!` methods for matrices sometimes seem not to get hit. Here is a demonstration:

```julia
using LinearAlgebra, SparseArrays, LimitedLDLFactorizations

n  = 100
S  = sprandn(n, n, 1/n) + 100*I
pr = lldl(S; memory=2)
v1 = randn(n)
v2 = randn(n, 10)

println("Trying vector ldiv!:")
ldiv!(pr, v1)
println("...done.")

println("Trying matrix ldiv!:")
ldiv!(pr, v2)
println("...done.")
```

On the current release, this is what you get when you run this program (also with a `versioninfo()` for your information):
```
cg:lldl> julia --project --quiet
julia> versioninfo()
Julia Version 1.11.5
Commit 760b2e5b739 (2025-04-14 06:53 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × 12th Gen Intel(R) Core(TM) i7-1260P
  WORD_SIZE: 64
  LLVM: libLLVM-16.0.6 (ORCJIT, alderlake)
Threads: 1 default, 0 interactive, 1 GC (on 12 virtual cores)
Environment:
  LD_LIBRARY_PATH = :/opt/knitro-13.2.0-Linux-64/lib:/opt/knitro-13.2.0-Linux-64/lib:/opt/knitro-13.2.0-Linux-64/lib

julia> include("testlldl.jl")
Trying vector ldiv!:
...done.
Trying matrix ldiv!:
ERROR: LoadError: MethodError: no method matching ldiv!(::LimitedLDLFactorization{Float64, Int64, Vector{Int64}, Vector{Int64}}, ::Matrix{Float64})
The function `ldiv!` exists, but no method is defined for this combination of argument types.

Closest candidates are:
  ldiv!(::Cholesky{T, <:StridedMatrix{T} where T}, ::StridedVecOrMat{T}) where T<:Union{Float32, Float64, ComplexF64, ComplexF32}
   @ LinearAlgebra ~/.julia/juliaup/julia-1.11.5+0.x64.linux.gnu/share/julia/stdlib/v1.11/LinearAlgebra/src/cholesky.jl:579
  ldiv!(::Cholesky, ::AbstractVecOrMat)
   @ LinearAlgebra ~/.julia/juliaup/julia-1.11.5+0.x64.linux.gnu/share/julia/stdlib/v1.11/LinearAlgebra/src/cholesky.jl:582
  ldiv!(::LinearAlgebra.AbstractQ, ::AbstractVecOrMat)
   @ LinearAlgebra ~/.julia/juliaup/julia-1.11.5+0.x64.linux.gnu/share/julia/stdlib/v1.11/LinearAlgebra/src/abstractq.jl:246
  ...

Stacktrace:
 [1] top-level scope
   @ ~/Scratch/lldl/testlldl.jl:15
 [2] include(fname::String)
   @ Main ./sysimg.jl:38
 [3] top-level scope
   @ REPL[2]:1
in expression starting at /home/cg/Scratch/lldl/testlldl.jl:15
```
So it seems like `ldiv!` for an LLDL object and a `Matrix` is not hitting the special branch.

This PR makes just a few small tweaks to fix this. First, I split your special `ldiv!` methods into non-exported `_ldiv!`, which at least personally I find preferable so that if dispatch breaks or something I can at least import an internal method and not have to drop what I'm doing to catch something. Second, the code previously defined `ldiv!` for both `AbstractVector` and `AbstractMatrix` separately. But that dispatch is handled by the methods of `lldl_solve!`, so I don't think the different `ldiv!`s are necessary and I removed them.

The tests and everything still pass. Is there any other way that you'd like me to kick the tires on the change? Or any other formatting preferences you'd like this PR to adhere to?

amontoison reviewed (changes requested):
@cgeoga 
Can you add a unit tests where `ldiv!` is called with a matrix `B` as right-hand side?

amontoison commented:
Can you also rebase your branch?
I fixed CI on `main`.

cgeoga commented:
@amontoison, thanks very much for taking a look! I think I have rebased successfully (although I really don't know git very well, so please excuse me if this is not correct). I have also added a test with `ldiv!` for matrix RHS args. Interestingly there were some other tests that that I would have expected to hit that error as well, but maybe they weren't (or is that just because the CI was broken?). In any case, let me know if this looks good!

amontoison commented:
Thanks @cgeoga !
I will tag a new release.
